public/app/features/alerting/unified/hooks/useProduceNewRuleGroup.test.tsx

```diff
@@ -0,0 +1,48 @@
+import { waitFor, renderHook } from 'test/test-utils';
+
+import { setBackendSrv } from '@grafana/runtime';
+import { backendSrv } from 'app/core/services/backend_srv';
+import { RuleGroupIdentifier } from 'app/types/unified-alerting';
+
+import { setupMswServer } from '../mockApi';
+import { grafanaRulerGroupName, grafanaRulerNamespace, grafanaRulerRule } from '../mocks/alertRuleApi';
+import { GRAFANA_RULES_SOURCE_NAME } from '../utils/datasource';
+
+import { usePauseRuleInGroup } from './useProduceNewRuleGroup';
+
+setupMswServer();
+
+beforeAll(() => {
+  setBackendSrv(backendSrv);
+});
+
+it('should be able to pause a rule', async () => {
+  const ruleGroupIdentifier: RuleGroupIdentifier = {
+    namespace: grafanaRulerNamespace.uid,
+    group: grafanaRulerGroupName,
+    ruleSourceName: GRAFANA_RULES_SOURCE_NAME,
+  };
+
+  expect(grafanaRulerRule).toHaveProperty('grafana_alert.is_paused', false);
+
+  const { result } = renderHook(() => {
+    const [pauseRule, pauseState] = usePauseRuleInGroup();
+    pauseRule(ruleGroupIdentifier, grafanaRulerRule, true);
+
+    return pauseState;
+  });
+
+  await waitFor(() => {
+    expect(result.current).toHaveProperty('status', 'fulfilled');
+  });
+
+  // a bit hacky to inspect what was passed in to RTKQ – open to ideas
+  expect(result.current).toHaveProperty('originalArgs.payload.rules[0].grafana_alert.is_paused', true);
```

Comment:
Could use (deprecated) `waitForServerRequest`, or we continue moving towards making the mock server more stateful/behave as we expect. 

i.e., when mock server receives new rule group payload, it updates its state for that rule group, so if you were to request it again you get what the new group is (like the real API would behave)

So you could test by using the hook to make a call to pause a rule, and use another hook/even just directly `fetch` the endpoint you know is responsible (which MSW would handle), and check that the response from the mock server still contains all the rules you're expecting. Or we separately then render the rule list/rule detail view and assert that it says "paused"